### PR TITLE
Move CI to Github Actions

### PR DESCRIPTION
**Pull request by HolyPrapor**

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -2,38 +2,41 @@ name: CMake
...
+      
+    - name: Prepare for release
+      run: tar -zcf odyssey.linux-amd64.$(git rev-parse --short HEAD).tar.gz sources
+
+    - name: Release
+      uses: fnkr/github-action-ghr@v1
+      if: startsWith(github.ref, 'refs/tags/') && matrix.os == 'ubuntu-18.04' && matrix.compiler == 'gcc' # We need to release only once
+      env:
+        GHR_PATH: odyssey.linux-amd64.$(git rev-parse --short HEAD).tar.gz
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (HolyPrapor) [inline .github/workflows/cmake.yml:42]: Will work only after filling `secrets` of repository.

See more info [there](https://docs.github.com/en/actions/reference/encrypted-secrets).

File: .github/workflows/coverity.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+on:
+  push:
+    branches: [ coverity_scan ]
+
+jobs:
+  scan:
+    runs-on: ubuntu-18.04
+    env:
+      TOKEN: ${{ secrets.COVERITY_SCAN_TOKEN }}
```
Comment (HolyPrapor) [inline .github/workflows/coverity.yml:11]: Will work only after filling `secrets` of repository.

See more info [there](https://docs.github.com/en/actions/reference/encrypted-secrets).

**Comment by HolyPrapor** [comment]
Unfortunately, I'm not able to test `coverity` workflow because I do not possess COVERITY_SCAN_TOKEN.

File: .github/workflows/cmake.yml [hunk 3]
```diff
@@ -2,38 +2,41 @@ name: CMake
...
+        
       
-    - name: Build
-      working-directory: ${{runner.workspace}}/build
-      shell: bash
-      # Execute the build.  You can specify a specific target with "--target <NAME>"
-      run: make local_build BUILD_TEST_DIR=$GITHUB_WORKSPACE
-
+    - name: Format
+      continue-on-error: true
```
Comment (HolyPrapor) [inline .github/workflows/cmake.yml:25]: This can be safely removed. I put it in order to evade failing style check.

**Comment by HolyPrapor** [comment]
You can also see [there](https://github.com/HolyPrapor/odyssey/actions) that these workflows are correct.

**Comment by reshke** [comment]
Hi! thanks for the job done, lets try

**Pull request merged**